tests/test_configure.py

```diff
@@ -1254,6 +1261,20 @@ def test_send_to_logfire_if_token_present_empty() -> None:
         del os.environ['LOGFIRE_TOKEN']
 
 
+def test_send_to_logfire_if_token_present_empty_via_env_var() -> None:
+    os.environ['LOGFIRE_TOKEN'] = ''
+    os.environ['LOGFIRE_SEND_TO_LOGFIRE'] = 'if-token-present'
+    try:
+        with ExitStack() as stack:
+            stack.enter_context(mock.patch('logfire._internal.config.Confirm.ask', side_effect=RuntimeError))
+            requests_mocker = stack.enter_context(requests_mock.Mocker())
+            configure(console=False)
+            assert len(requests_mocker.request_history) == 0
+    finally:
+        del os.environ['LOGFIRE_TOKEN']
+        del os.environ['LOGFIRE_SEND_TO_LOGFIRE']
```

Comment:
```suggestion
    with patch.dict(
        os.environ,
        {'LOGFIRE_TOKEN': '', 'LOGFIRE_SEND_TO_LOGFIRE': 'if-token-present'},
    ), mock.patch(
        'logfire._internal.config.Confirm.ask',
        side_effect=RuntimeError,
    ), requests_mock.Mocker() as requests_mocker:
        configure(console=False)
    assert len(requests_mocker.request_history) == 0
```

(i'm not sure that this is better than the exit stack, i started writing it before i realised that i couldn't make it pretty)